lib/db_ido_pgsql/idopgsqlconnection.cpp

```diff
@@ -700,9 +700,15 @@ bool IdoPgsqlConnection::FieldToEscapedString(const String& key, const Value& va
 
 		*result = static_cast<long>(dbrefcol);
 	} else if (DbValue::IsTimestamp(value)) {
-		long ts = rawvalue;
+		double ts = rawvalue;
+
+		if (ts < -2e11 || ts > 9e12) {
```

Comment:
How were these bounds chosen? There should be a comment explaining where these magic numbers come from.

Reply:
Just use the PostgreSQL limit please (and document it).

Reply:
* 💡 What about four year digits instead, like #10059

> // RFC 3339: "All dates and times are assumed to be (...) somewhere between 0000AD and 9999AD."

(Also, otherwise the IDO would be much more capable than Icinga DB 😅)

Just an idea. I'm also fine with what Eric said. (https://github.com/Icinga/icinga2/pull/10058#discussion_r1772906869)

Reply:
I don't care too much about the exact limit as long as there's a reasonable explanation for it (hard limit of the database, limit of a common representation format). But I don't see why we should introduce an arbitrary new limit just because it has a nicer decimal representation that's not relevant anywhere.[^1] After all, if at any time a timestamp more than a few years in the future appears, chances are high that it's unintentional.

[^1]: Yes, 9999 (or less than 10000) would also be a choice based on the decimal representation, but in this case of a value that's commonly used, even in machine-readable representations where it's often limited to 4 digits, so there's a justification for that limit.